components/core/src/clp_s/ArchiveWriter.cpp

```diff
@@ -67,6 +92,113 @@ void ArchiveWriter::close() {
+    compressor.write_numeric_value(ArchiveMetadataPacketType::ArchiveInfo);
+    compressor.write_numeric_value(archive_info_str.size());
+    compressor.write_string(archive_info_str);
+
+    // Write archive file info
+    ArchiveFileInfoPacket archive_file_info{.files{files}};
+    msgpack_buffer.clear();
+    msgpack::pack(msgpack_buffer, archive_file_info);
+    std::string archive_file_info_str = msgpack_buffer.str();
+    compressor.write_numeric_value(ArchiveMetadataPacketType::ArchiveFileInfo);
+    compressor.write_numeric_value(archive_file_info_str.size());
+    compressor.write_string(archive_file_info_str);
+
+    // Write timestamp dictionary
+    compressor.write_numeric_value(ArchiveMetadataPacketType::TimestampDictionary);
+    compressor.write_numeric_value(timestamp_dict_compressed_size);
+    std::string timestamp_dict_path = m_archive_path + constants::cArchiveTimestampDictFile;
+    FileReader timestamp_dict_reader;
+    timestamp_dict_reader.open(timestamp_dict_path);
+    char read_buffer[cReadBlockSize];
+    while (true) {
+        size_t num_bytes_read{0};
+        ErrorCode error_code
+                = timestamp_dict_reader.try_read(read_buffer, cReadBlockSize, num_bytes_read);
+        if (ErrorCodeSuccess != error_code) {
+            break;
+        }
+        compressor.write(read_buffer, num_bytes_read);
+    }
+    timestamp_dict_reader.close();
+    std::filesystem::remove(timestamp_dict_path);
+    compressor.close();
+}
+
+void ArchiveWriter::write_archive_files(
+        FileWriter& archive_writer,
+        std::vector<ArchiveFileInfo> const& files
+) {
+    for (auto const& file : files) {
+        std::string file_path = m_archive_path + file.n;
+        FileReader reader;
+        reader.open(file_path);
+        char read_buffer[cReadBlockSize];
+        while (true) {
+            size_t num_bytes_read{0};
+            ErrorCode const error_code
+                    = reader.try_read(read_buffer, cReadBlockSize, num_bytes_read);
+            if (ErrorCodeSuccess != error_code) {
+                break;
+            }
+            archive_writer.write(read_buffer, num_bytes_read);
+        }
+        reader.close();
+        boost::filesystem::remove(file_path);
+    }
+}
+
+void ArchiveWriter::write_archive_header(FileWriter& archive_writer, size_t metadata_section_size) {
+    ArchiveHeader header{
+            .magic_number{0},
+            .version
+            = (cArchiveMajorVersion << 24) | (cArchiveMinorVersion << 16) | cArchivePatchVersion,
+            .uncompressed_size = m_uncompressed_size,
+            .compressed_size = m_compressed_size,
+            .reserved_padding{0},
+            .metadata_section_size = static_cast<uint32_t>(metadata_section_size),
+            .compression_type = static_cast<uint16_t>(ArchiveCompressionType::Zstd),
+            .padding = 0
+    };
+    std::memcpy(&header.magic_number, "ARCHIVES", sizeof(header.magic_number));
```

Comment:
Just realizing that I forgot to actually specify a magic number in the spec. We should pick one and add it to the spec doc. Maybe use something close to the magic numbers for IR streams (from src/clp/ffi/ir_stream/protcol_constants.hpp)?